### PR TITLE
refactor: Do not call teardown manually in tests

### PR DESCRIPTION
**Pull request by alexandear**
The PR refactors the [`setup`](https://github.com/google/go-github/blob/2406067865f4b36a5b994868e4bc3f1922bf8a32/github/github_test.go#L37) function by passing `*testing.T` parameter and calling `t.Cleanup(server.Close)` inside. So a test that using `setup` doesn't need to call manually `teardown`.

This leads to removal of 1319 code lines in tests.

### CODE REVIEW

File: scrape/scrape_test.go
Comment (gmlewis) [inline scrape/scrape_test.go:23]:
```suggestion
	t.Cleanup(server.Close)

	return client, mux
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @alexandear !
One important change, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: scrape/scrape_test.go
Comment (alexandear) [reply scrape/scrape_test.go:25]: Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @alexandear !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request closed without merging**
